File: backend/banking-service/src/main/java/com/finix/transaction/service/TransactionService.java
package com.finix.transaction.service;

import java.time.LocalDate;
import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import com.finix.transaction.dto.GetAllTransactionsDto;
import com.finix.transaction.dto.TransferTransactionDto;
import com.finix.transaction.entity.TransactionNature;
import com.finix.transaction.entity.TransactionStatus;

@Service
@Transactional
public interface TransactionService {

	ResponseEntity<?> transferMoney(TransferTransactionDto transactionDto);


	Page<GetAllTransactionsDto> getTransactions(int page, int size, String sortBy, String direction,
			TransactionNature nature, TransactionStatus status, LocalDate fromDate, LocalDate toDate);
	
}

File: frontend/src/services/transactionService.js
import api from './api';

export const transactionService = {
  /**
   * Perform money transfer
   * @param {Object} data - { accountType, toAccount, amount, referenceNumber, remarks }
   */
  async transferMoney(data) {
    const response = await api.post('/api/transaction', data);
    return response.data; // Plain string message response
  },

  /**
   * Get paginated & filterable transaction history
   * @param {Object} params - { page, size, sortBy, direction, nature, status, fromDate, toDate }
   */
  async getTransactions(params = {}) {
    const response = await api.get('/api/transaction', { params });
    return response.data;
  },
};
